SME.CDEP.Aplicacao/Servicos/ServicoFormato.cs

```diff
@@ -0,0 +1,50 @@
+﻿using AutoMapper;
+using SME.CDEP.Aplicacao.DTOS;
+using SME.CDEP.Aplicacao.Servicos.Interface;
+using SME.CDEP.Dominio.Entidades;
+using SME.CDEP.Infra.Dados.Repositorios.Interfaces;
+
+namespace SME.CDEP.Aplicacao.Servicos
+{
+    public class ServicoFormato : IServicoFormato
+    {
+        private readonly IRepositorioFormato repositorioFormato;
+        private readonly IMapper mapper;
+        
+        public ServicoFormato(IRepositorioFormato repositorioFormato, IMapper mapper) 
+        {
+            this.repositorioFormato = repositorioFormato ?? throw new ArgumentNullException(nameof(repositorioFormato));
+            this.mapper = mapper ?? throw new ArgumentNullException(nameof(mapper));
+        }
+
+        public async Task<long> Inserir(FormatoDTO formatoDto)
+        {
+            var formato = mapper.Map<Formato>(formatoDto);
+            return await repositorioFormato.Inserir(formato);
+        }
+
+        public async Task<IList<FormatoDTO>> ObterTodos()
+        {
+            return (await repositorioFormato.ObterTodos()).ToList().Where(w=> !w.Excluido).Select(s=> mapper.Map<FormatoDTO>(s)).ToList();
```

Comment:
o primeiro .ToList() não faz sentido pois o where e select geram um IEnumerable novamente.
```suggestion
        public async Task<IList<FormatoDTO>> ObterTodos()
        {
            return (await repositorioFormato.ObterTodos()).Where(w=> !w.Excluido).Select(s=> mapper.Map<FormatoDTO>(s)).ToList();
```